tox.ini

```diff
@@ -0,0 +1,62 @@
+[tox]
+envlist = py{35,36,37},py{35,36,37}-alldeps,docs,dev,pre,flake8
+skipsdist = True
+requires = tox-venv
+           setuptools >= 30.0.0
+
+[testenv]
+usedevelop = True
+whitelist_externals = which
+passenv = HOME
+deps =
+    !alldeps: cython
+    !alldeps: jinja2
+    !alldeps: pytest-astropy
+    pytest-cov
+    !alldeps: pytest-xdist
+    py35-!alldeps: numpy==1.13.3
+    py{36,37}-!alldeps: numpy>=1.13
+    alldeps: -rpip-requirements-dev
```

Comment:
I personally find it quite hard to read the previous 8-9 lines - even though it might be a bit more verbose, what about writing them out in separate tox.ini entries? (like for the builds below).

Reply:
The goal was to avoid declaring dependencies multiple times, especially for optional ones. But as pip will complain if a dependency is included twice, and pip-requirements include numpy/cython, it is annoying to manage. 
But all this should be easier with your setup_requires PR!
Also I removed the numpy pinning with the upcoming commits.